File: gym/102007/c/c.cpp
#include <bits/stdc++.h>
using namespace std;

constexpr int Ma = 1e6 + 100, inf = 0x3f3f3f3f;
using ll = long long;
#define ep emplace_back
template<typename T>
bool cmin(T& a, const T& b) {return a <= b || (a = b, false);}

ll f[Ma];

int main() {
    memset(f, inf, sizeof f);
    int n; scanf("%d", &n);
    for (int i = 1; i < Ma; i++)
        for (int j = i; j < Ma; j += i) {
            cmin(f[j], (ll)i + j / i);
        }
    ll ans = numeric_limits<ll>::max();
    for (int i = 1; i <= n; i++) if (n % i == 0) {
        cmin(ans, (ll)n / i * 2 + f[n / i] * i * 2);
    }
    printf("%lld\n", ans);
}

File: gym/102007/c/c.py
n = int(input())
ans = float('inf')
g = [[] for _ in range(1000001)]
for i in range(1, 1000001) :
    for j in range(i, 1000001, i) :
        g[j].append(i)
for i in range(1, n + 1) :
    if n % i == 0 :
        mi = float('inf')
        for x in g[n // i] :
            mi = min(mi, x + n // i // x)
        ans = min(ans, n // i * 2 + mi * i * 2)
print(ans)

